packages/core/src/routes/interaction/verifications/profile-verification.ts

```diff
@@ -164,63 +169,64 @@ const profileExistValidation = async (
   }
 };
 
-// eslint-disable-next-line complexity
+const isValidRegisterProfile = (profile: Profile): profile is RegisterSafeProfile =>
+  registerProfileSafeGuard.safeParse(profile).success;
+
+const hasPasswordInProfile = (profile: Profile): profile is { password: string } =>
+  Boolean(profile.password);
```

Comment:
use `'password' in profile`?

Reply:
the reason is there is no password-required type Profile defined, so have to specify the type manually here.

Reply:
either way works, keep it simple here.

Reply:
updated per out offline discuss please check @charIeszhao